In [6]:
import pandas as pd 

# Read CSV file into a Dataframe
df = pd.read_csv("titanic.csv") 

# Show the first 5 rows 
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

# What percentage of female passengers are likely to survive?
This is an observational study 
I will be writing a code figure out the percentage from the data table 
Use the data table to figure out the percentage 

In [5]:
import pandas as pd 

# Read CSV file into a Dataframe
df = pd.read_csv("titanic.csv") 

# Show the first 7 rows 
print(df.head()) 

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


In [12]:
import pandas as pd 

# Read CSV file into a Dataframe
df = pd.read_csv("titanic.csv") 

# Get the 7th row (index 8 since python is 0 based) 
seventh_sample = df.iloc[8] 
print(seventh_sample) 

# If you want it as a Dataframe instead os a series : 
seventh_sample_df = df.iloc[8] 
seventh_sample

PassengerId                                                    9
Survived                                                       1
Pclass                                                         3
Name           Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)
Sex                                                       female
Age                                                         27.0
SibSp                                                          0
Parch                                                          2
Ticket                                                    347742
Fare                                                     11.1333
Cabin                                                        NaN
Embarked                                                       S
Name: 8, dtype: object


PassengerId                                                    9
Survived                                                       1
Pclass                                                         3
Name           Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)
Sex                                                       female
Age                                                         27.0
SibSp                                                          0
Parch                                                          2
Ticket                                                    347742
Fare                                                     11.1333
Cabin                                                        NaN
Embarked                                                       S
Name: 8, dtype: object

In [17]:
df[df["Embarked"] == "Q" ]

PassengerId  Survived  Pclass                                  Name  \
5              6         0       3                      Moran, Mr. James   
16            17         0       3                  Rice, Master. Eugene   
22            23         1       3           McGowan, Miss. Anna "Annie"   
28            29         1       3         O'Dwyer, Miss. Ellen "Nellie"   
32            33         1       3              Glynn, Miss. Mary Agatha   
..           ...       ...     ...                                   ...   
790          791         0       3              Keane, Mr. Andrew "Andy"   
825          826         0       3                       Flynn, Mr. John   
828          829         1       3          McCormack, Mr. Thomas Joseph   
885          886         0       3  Rice, Mrs. William (Margaret Norton)   
890          891         0       3                   Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
5      male   NaN      0      0  330877   8.4583   NaN        Q  
16     male   2.0      4      1  382652  29.1250   NaN        Q  
22   female  15.0      0      0  330923   8.0292   NaN        Q  
28   female   NaN      0      0  330959   7.8792   NaN        Q  
32   female   NaN      0      0  335677   7.7500   NaN        Q  
..      ...   ...    ...    ...     ...      ...   ...      ...  
790    male   NaN      0      0   12460   7.7500   NaN        Q  
825    male   NaN      0      0  368323   6.9500   NaN        Q  
828    male   NaN      0      0  367228   7.7500   NaN        Q  
885  female  39.0      0      5  382652  29.1250   NaN        Q  
890    male  32.0      0      0  370376   7.7500   NaN        Q  

[77 rows x 12 columns]

In [16]:
df.iloc [14]

PassengerId                                      15
Survived                                          0
Pclass                                            3
Name           Vestrom, Miss. Hulda Amanda Adolfina
Sex                                          female
Age                                            14.0
SibSp                                             0
Parch                                             0
Ticket                                       350406
Fare                                         7.8542
Cabin                                           NaN
Embarked                                          S
Name: 14, dtype: object

In [1]:
import pandas as pd

df = pd.read_csv('titanic.csv') 
female_passengers = df[df['Sex'] == 'female']
total_females = len(female_passengers)
female_survivors = len(female_passengers[female_passengers['Survived'] == 1])
female_survival_percentage = (female_survivors / total_females) * 100
print(f"Percentage of female passengers who survived: {female_survival_percentage:.2f}%")


Percentage of female passengers who survived: 74.20%


# Analysis

About 74 out of every 100 women on the Titanic survived, which is a lot higher than the overall survival rate. This happened mostly because of the “women and children first” rule during the rescue, where women got priority for lifeboats. Back then, society believed in protecting women, so this was a big reason why so many of them made it through. Basically, thanks to that rule and the way things worked at the time, women had a much better shot at survival.

# What percentage of female passengers are likely to survive?

About 74.2% of women population on the titanic survived are most likely to survive which means 2 out of every 3 wowmens survived. Out of 314 female passengers, 233 lived through the disaster. This survival rate also shows that women had a much better chance of surviving then the mens on the titanic. 